android/src/main/java/com/marigold/rnsdk/RNMessageStreamModule.kt

```diff
@@ -27,20 +31,45 @@ class RNMessageStreamModule (reactContext: ReactApplicationContext, private val
     @VisibleForTesting
     internal var jsonConverter: JsonConverter = JsonConverter()
 
+    private val eventListener = EventListener()
+    private var showDefaultInAppNotification = true
+
     init {
         messageStream.setOnInAppNotificationDisplayListener(this)
     }
 
     override fun shouldPresentInAppNotification(message: Message): Boolean {
-        try {
-            val writableMap = jsonConverter.convertJsonToMap(message.toJSON())
-            reactApplicationContext
+        return runBlocking {
```

Comment:
We should add a flag to allow the app to opt-in to this behaviour. Add a check on the flag at the start here and just return true if the calling app never intends to handle the in-app notifications.